File: Python/Medium/Sorting/1727__Largest_Submatrix_With_Rearrangements.py
def largestSubmatrix(matrix: list[list[int]]) -> int:
    m, n = len(matrix), len(matrix[0])
    for i in range(1, m):
        for j in range(n):
            if matrix[i][j] == 1:
                matrix[i][j] += matrix[i - 1][j]
    max_area = 0

    for row in matrix:
        row.sort(reverse=True)
        for j in range(n):
            max_area = max(max_area, row[j] * (j + 1))
    return max_area


matrix = [[0, 0, 1], [1, 1, 1], [1, 0, 1]]
print(largestSubmatrix(matrix))

File: PHP/Medium/Sorting/1727__Largest_Submatrix_With_Rearrangements.php
<?php
function largestSubmatrix($matrix)
{
    $m = count($matrix);
    $n = count($matrix[0]);

    for ($i = 1; $i < $m; $i++) {
        for ($j = 0; $j < $n; $j++) {
            if ($matrix[$i][$j] == 1) {
                $matrix[$i][$j] += $matrix[$i - 1][$j];
            }
        }
    }
    $max_area = 0;
    foreach ($matrix as &$row) {
        rsort($row);
        for ($j = 0; $j < $n; $j++) {
            $max_area = max($max_area, $row[$j] * ($j + 1));
        }
    }
    return $max_area;
}

echo largestSubmatrix([[0, 0, 1], [1, 1, 1], [1, 0, 1]]);
echo largestSubmatrix([[1, 1, 0], [1, 0, 1]]);